vendor/github.com/openshift/cluster-api/pkg/controller/machine/controller.go

```diff
@@ -145,6 +163,51 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 			return reconcile.Result{}, nil
 		}
 		klog.Infof("reconciling machine object %v triggers delete.", name)
+
+		// Drain node before deletion
```

Comment:
We should think about breaking this large function apart, it's going to be very difficult if not impossible to test.

Reply:
there are not unit tests so far other than the ones provided by https://github.com/openshift/kubernetes-drain/. From the functional point of view we have https://github.com/openshift/cluster-api-actuator-pkg/blob/master/pkg/e2e/actuators/actuators.go#L182.